src/cpu/aarch64/jit_uni_reorder.cpp

```diff
@@ -161,13 +161,18 @@ struct jit_uni_reorder_kernel_f32_t : public kernel_t, public jit_generator {
     static bool applicable(const prb_t &p) {
         using namespace data_type;
 
+        bool bf16_ok = (mayiuse_bf16() && (p.itype == bf16) && (p.otype == bf16)
+                               && !interim_f32_needed(p, false))
+                || (p.itype != bf16 && p.otype != bf16)
+                || (p.itype == f32 && mayiuse_bf16());
+
         bool ok = true && p.ndims > 0
-                && utils::one_of(p.itype, f32, s32, data_type::s8, u8)
+                && utils::one_of(p.itype, f32, bf16, s32, data_type::s8, u8)
                 && utils::one_of(p.otype, f32, bf16, s32, data_type::s8, u8)
                 && utils::everyone_is(0, p.ioff, p.ooff) /* do we need this? */
                 && utils::one_of(p.beta, 0.f, 1.f) /* anything else? */
                 && simple_impl_desc_init(p, nullptr) && prb_has_small_strides(p)
-                && ((p.otype != bf16) || (p.itype == f32 && mayiuse_bf16()));
+                && bf16_ok;
```

Comment:
```suggestion
                && IMPLICATION(utils::one_of(bf16, p.itype, p.otype), mayiuse_bf16())
```
I'm not not sure only about `p.beta == 1.f` case, since it's not clear if it's supported for bf16 and whether it impacts the code.

Reply:
@Ryo-not-rio, could you comment on this one, please?

Reply:
Hi, yes I've looked into it and it looks like bf16 is not supported for `p.beta == 1.f` case like you said and a patch is on it's way to fix this

Reply:
I have created a [new PR](https://github.com/oneapi-src/oneDNN/pull/1915) addressing this comment so this PR can be closed